### ISSUE TITLE
Having trouble importing epub

### ISSUE THREAD

**Issue by nashihang777** [tvhong/incremental-reading#37]
I encountered a "not exist" error when importing some epub files, while other epub files did not report an error. i don't konw why

![image_1732698052951_0](https://github.com/user-attachments/assets/f90523d3-bd77-4bac-8244-cbd3f3b4d973)

**Comment by tvhong** [comment]
Do you have some example epub files that you can share? Ideally, at least 1 success and 1 failure file.

**Comment by nashihang777** [comment]
Anki 24.11 (87ccd24e)  (ao)
Python 3.9.18 Qt 6.6.2 PyQt 6.6.1
Platform: Windows-10-10.0.19041

===Add-ons (active)===
(add-on provided name [Add-on folder, installed at, version, is config changed])
Advanced Browser ['874215009', 2024-10-27T22:20, 'None', '']
AnkiConnect ['2055492159', 2024-11-07T09:34, 'None', mod]
Edit Field During Review Cloze ['385888438', 2024-11-16T18:23, '6.21', mod]
Enhanced Cloze for Anki 21 ['1990296174', 2023-09-11T02:48, 'None', '']
FSRS Helper Postpone  Advance  Load Balance  Easy Days  Disperse Siblings ['759844606', 2024-12-07T19:00, 'None', '']
Image Occlusion Enhanced ['1374772155', 2022-04-09T15:15, 'None', '']
Incremental Reading v4119 unofficial clone ['999215520', 2023-11-23T01:54, 'None', '']

===IDs of active AnkiWeb add-ons===
1374772155 1990296174 2055492159 385888438 759844606 874215009 999215520

===Add-ons (inactive)===
(add-on provided name [Add-on folder, installed at, version, is config changed])

[Uploading epub.zip…]()

**Comment by tvhong** [comment]
Thank you for providing the sample files!

I just tried to import all 3 of them in my Mac machine and they all succeeded. I suspect this might be something Windows specific.

Can you try accessing the file that were reported as "Not exists" and see if it's there?

I also see this part of the file name that doesn't exist in the file you sent me. What if you remove all special characters (e.g. `(`, `)`, `.`) in your epub file?
<img width="397" alt="image" src="https://github.com/user-attachments/assets/0f310e09-7a55-4f76-ba3d-376ea1ef9047" />

**Comment by nashihang777** [comment]
I deleted the special characters in the file name and still got an error. In addition, those files that were reported as not existing do indeed exist in the directory indicated by the bug.
![anki_bug](https://github.com/user-attachments/assets/9d5338ee-717b-4df6-9be0-e2fd8be0fc97)
![anki_bug2](https://github.com/user-attachments/assets/d03bbc6a-f951-443b-969c-c582b94f157f)
![anki_bug3](https://github.com/user-attachments/assets/749b3a16-5bc8-40ab-b23f-4fb0ea0638b9)

**Comment by tvhong** [comment]
I just did some investigation and suspect that the addon was looking at the wrong drive (see how the path doesn't include a drive letter?).

Q: Does it successfully import a any articles/chapter in the failure file or did it just fail immediately?

**Comment by nashihang777** [comment]
Perhaps you're right; the plugin might be making a mistake in path recognition. When I import the bug file, the plugin first provides a list of split chapters, but no matter what I select, it ultimately results in an error after I click confirm.

**Comment by tvhong** [comment]
Did any of the selected chapters get successfully imported?

**Comment by nashihang777** [comment]
Never managed to successfully import the selected chapters; after the bug appeared, the entire Anki interaction interface froze, and I had to use the Windows Task Manager to close Anki. I carefully considered the path issue you mentioned and recalled that I had once changed my Win10 system's Documents directory from the C drive to the same path under a different drive letter. Perhaps this is the reason for the bug?

**Comment by tvhong** [comment]
Yeah, that could be related. I think this has something to do with the default drive that the add-on assumes, and the drive where the content of the epub is extracted.

Also, the fact that you couldn't import any chapters means that the problem is likely at the folder level and not a per-file problem.

I have an idea for a fix that I can try in the next release. It might fix your problem (can't confirm as I can't reproduce).

**Comment by tvhong** [comment]
@nashihang777 I tried to fix this in 4.12.1. If you can try it out and let me know how it goes, that'd be appreciated.

**Comment by nashihang777** [comment]
It's unfortunate that an error still occurred, and the error message is exactly the same as before. As a result, I can only close Anki through the Windows Task Manager.

Anki 24.11 (87ccd24e)  (ao)
Python 3.9.18 Qt 6.6.2 PyQt 6.6.1
Platform: Windows-10-10.0.19041

===Add-ons (active)===
(add-on provided name [Add-on folder, installed at, version, is config changed])
Advanced Browser ['874215009', 2024-10-27T22:20, 'None', '']
AnkiConnect ['2055492159', 2024-11-07T09:34, 'None', mod]
Edit Field During Review Cloze ['385888438', 2024-11-16T18:23, '6.21', mod]
Enhanced Cloze for Anki 21 ['1990296174', 2023-09-11T02:48, 'None', '']
FSRS Helper Postpone  Advance  Load Balance  Easy Days  Disperse Siblings ['759844606', 2025-01-10T15:28, 'None', '']
Image Occlusion Enhanced ['1374772155', 2022-04-09T15:15, 'None', '']
Incremental Reading v4121 unofficial clone ['999215520', 2025-01-14T05:27, 'None', '']

===IDs of active AnkiWeb add-ons===
1374772155 1990296174 2055492159 385888438 759844606 874215009 999215520

===Add-ons (inactive)===
(add-on provided name [Add-on folder, installed at, version, is config changed])

![屏幕截图 2025-01-14 135027](https://github.com/user-attachments/assets/8586d61f-059e-4a84-9a5c-a8c0f84d1f79)

**Comment by tvhong** [comment]
Hm.. how comfortable are you with using a terminal to launch Anki and make changes to Anki add-on code (or I can send you a zip file as well)?

I think I need some logs to debug this further.

**Comment by nashihang777** [comment]
I can try to launch Anki from the terminal and send you the log. Do you have WeChat? It would make our communication more convenient.
